utils/common/src/main/java/org/apache/brooklyn/util/guava/Maybe.java

```diff
@@ -457,6 +457,8 @@ public boolean equals(Object obj) {
         Maybe<?> other = (Maybe<?>)obj;
         if (!isPresent()) 
             return !other.isPresent();
+        if (!other.isPresent())
+            return false;
```

Comment:
Don't mind me, somehow missed the previous check where two absents will be equal.
My impression was exactly the opposite, that with the change two absents are never equal, even for `this.equals(this)`.